content/en/docs/tasks/administer-cluster/controller-manager-leader-migration.md

```diff
@@ -0,0 +1,109 @@
+
+Leader Migration provides a mechanism in which HA clusters can safely migrate "cloud specific" controllers between the `kube-controller-manager` and the `cloud-controller-manager` via a shared resource lock between the two components.
+
+This guide walks you through the process of upgrading the control plane from `kube-controller-manager` with built-in cloud provider to running both `kube-controller-manager` and `cloud-controller-manager`.
+
+## {{% heading "prerequisites" %}}
+
+The control plane nodes should run `kube-controller-manager` with Leader Election enabled through `--leader-elect=true`, and the cloud privider set with `--cloud-provider` flag. At this moment, `cloud-controller-manager` should not have been deployed.
+
+<!-- steps -->
+
+### Grant Access to Migration Lease
+
+The default premissions of controller manager allow only access to their main lease. In order for the migration to work, access to another lease is required.
+
+We can grant full access of leases API to `kube-controller-manager` by modifying `system::leader-locking-kube-controller-manager` role. In this guide, we assume the name of the migration lease is `cloud-provider-extraction-migration`.
+
+`kubectl patch -n kube-system role 'system::leader-locking-kube-controller-manager' -p '{"rules": [ {"apiGroups":[ "coordination.k8s.io"], "resources": ["leases"], "resourceNames": ["cloud-provider-extraction-migration"], "verbs": ["create", "list", "get", "update"] } ]}' --type=merge`
+
+Do the same to `system::leader-locking-cloud-controller-manager` role.
+
+`kubectl patch -n kube-system role 'system::leader-locking-cloud-controller-manager' -p '{"rules": [ {"apiGroups":[ "coordination.k8s.io"], "resources": ["leases"], "resourceNames": ["cloud-provider-extraction-migration"], "verbs": ["create", "list", "get", "update"] } ]}' --type=merge`
+
+### Initial Leader Migration Configuration
+
+Leader Migration requires a configuration file representing the state of controller-to-manager assignment. The following example configuration shows the assignment at this moment where `route`, `service`, and `cloud-node-lifecycle` controllers are started within `kube-controller-manager`.
+
+```yaml
+kind: LeaderMigrationConfiguration
+apiVersion: controllermanager.config.k8s.io/v1alpha1
+leaderName: cloud-provider-extraction-migration
+resourceLock: leases
+controllerLeaders:
+  - name: route
+    component: kube-controller-manager
+  - name: service
+    component: kube-controller-manager
+  - name: cloud-node-lifecycle
+    component: kube-controller-manager
+
+```
+
+On each control plane node, save the content to `/etc/leadermigration.conf`, and update the manifest of `kube-controller-manager` so that the file is mounted inside the container at the same location. Also, update the same manifest to add the following arguments:
```

Comment:
"mounted inside the container at the same location"

Probably worth showing this in yaml as well-- it saves a bit of time and it minimizes changes of someone misunderstanding the step.

Reply:
Oh, also, not all cluster configurations use the term "manifest". Might be clearer to say call it something like the "Pod declaration for the `kube-controller-manager`"?

Reply:
Term "manifest" is defined by kubelet, as "Manifests are standard Pod definitions in JSON or YAML format in a specific directory." Ref: https://kubernetes.io/docs/tasks/configure-pod-container/static-pod/#configuration-files

Because for a control plane node, pods like kube-api-server, kube-controller-manager, etc, are started as static pods by kubelet, I think it is ok to use "manifest" here.

Reply:
Maybe mention “static pod” to make it clear we're assuming that the control plane components are run that way. The manifests for static pods are, indeed, manifests.

Reply:
Added to prerequisites.